jetty-infinispan/infinispan-common/src/main/java/org/eclipse/jetty/session/infinispan/InfinispanKeyBuilder.java

```diff
@@ -0,0 +1,31 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.session.infinispan;
+
+/**
+ *
+ *
+ */
+public class InfinispanKeyBuilder
+{
+    public static String build(String contextPath, String vhost, String id)
+    {
+        return contextPath + "_" + vhost + "_" + id;
```

Comment:
IS this really only for Infinispan? Look generally useful.

Reply:
The SessionContext.toString() does something similar, but I didn't want to rely on as generic a method as toString() to generate the key for infinispan - if toString() ever changed, then infinispan would be broken.